File: FE-customer/src/types/responses/product.response.ts
export interface ProductResponse {
  id: number;
  name: string;
  description: string;
  price: number;
  categoryId: number;
  categoryName: string;
  imageUrls: string[];
  status: "AVAILABLE" | "UNAVAILABLE";
  createdAt: string;
  updatedAt: string;
  wishlist: boolean;
  tags?: string[];
}

export interface CategoryResponse {
  id: number;
  name: string;
}

export interface ProductDetailResponse extends ProductResponse {
  description: string;
  status: "AVAILABLE" | "UNAVAILABLE";
  rating: number;
  reviewCount: number;
  tags: string[];
}
export interface CartItem extends ProductResponse {
  quantity: number;
  note?: string;
}
export interface PageResponse<T> {
  content: T[];
  pageable: {
    pageNumber: number;
    pageSize: number;
    sort: { sorted: boolean; unsorted: boolean; empty: boolean };
    offset: number;
    paged: boolean;
    unpaged: boolean;
  };
  totalPages: number;
  totalElements: number;
  last: boolean;
  number: number;
  size: number;
  numberOfElements: number;
  first: boolean;
  empty: boolean;
}

export interface ProductWithRatingResponse extends ProductResponse {
  averageRating: number;
  reviewCount: number;
}

File: BE/src/main/java/com/siupo/restaurant/dto/response/ProductResponse.java
package com.siupo.restaurant.dto.response;

import com.siupo.restaurant.dto.ReviewDTO;
import com.siupo.restaurant.enums.EProductStatus;
import lombok.Builder;
import lombok.Data;

import java.util.List;

@Data
@Builder
public class ProductResponse {
    private Long id;
    private String name;
    private String description;
    private Double price;
    private List<String> imageUrls;
    private Long categoryId;
    private String categoryName;
    private List<ReviewDTO> reviews;
    private Double rating;
    private Integer reviewCount;
    private List<String> tags;
    private EProductStatus status;
    private boolean isWishlist;
}